common/lib/xmodule/xmodule/modulestore/mongo/draft.py

```diff
@@ -728,11 +728,9 @@ def _internal_depth_first(item_location, is_root):
             bulk_record.dirty = True
             self.collection.remove({'_id': {'$in': to_be_deleted}})
 
-        if self.signal_handler and not bulk_record.active:
-            self.signal_handler.send("course_published", course_key=course_key)
-
         # Now it's been published, add the object to the courseware search index so that it appears in search results
-        CoursewareSearchIndexer.add_to_search_index(self, location)
+        if self.signal_handler:
+            self.signal_handler.send("course_published", course_key=location.course_key)
```

Comment:
I think we may need a new signal for `item_published` in this case... otherwise we are reindexing the whole course every time that we get a single item updated